### ISSUE TITLE
speed it up more

### ISSUE THREAD

**Issue by jospoortvliet** [nextcloud/nextcloud.com#193]
I did https://github.com/nextcloud/nextcloud.com/pull/191 to speed up loading the site. This optimized a lot of images and increased our Google speed score from 68 to 88 - which helps our google ranking. This is what is left to do:

![spectacle b20235](https://cloud.githubusercontent.com/assets/551757/19433950/3e18232e-9463-11e6-9134-e66eedb37a6d.png)

I guess we should put the css & javascript in the footer or something for that... @LukasReschke not urgent but when you have time, thoughts on this?

**Comment by Espina2** [comment]
We should have some scripts to do that automatically and minify the code to. Right? 

The javascript I think they have to be compiled in one file only, like the sass compile one css file. Its less request's to the server.

**Comment by jospoortvliet** [comment]
The website does all the minification and compilation automatically, yes, that's why the checkboxes for those items are there. But we load some stuff in the header, not lower on the page. Not sure if we have to...

**Comment by Espina2** [comment]
Yup we need that. But If are better we can download all the files and include them in the "main.js".
The same can be done with the css fonts I guess. 

I'm not expert in this area maybe anyone with more knowledge can advice better.

**Comment by jospoortvliet** [comment]
Yeah, that is what the old site did and it requires compiling stuff if you make changes, breaking the git workflow completely.

What we should do is to get rid of all the javascript in the head, put it in the footer. It is delaying the loading of the page and hurting us on Google (which punishes pages for that kind of stuff).

Also, we have to fix the website to make sure it works without javascript. So the reveal-on-scroll stuff, that either has to work without javascript, or it has to go...

**Comment by Espina2** [comment]
> What we should do is to get rid of all the javascript in the head, put it in the footer. It is delaying the loading of the page and hurting us on Google (which punishes pages for that kind of stuff).

We should not do that at all. The javascript files if make that it only download the javascript files after download all the HTML stuff and it will slows down instead of speed it up. 

For example in you page of secure the buttons doesn't work after you load all the page it slows it down. 

![screen shot 2016-10-24 at 11 03 45](https://cloud.githubusercontent.com/assets/11150206/19641662/916ee230-99d9-11e6-925f-13fec54dc586.png)

If you remove javascript you will break most of the things of bootstrap. For example a navigation use's javascript a pop up uses javascript a collapse tag uses javascript. Forget about hiding content and reveal them. This is are very step forward for me.

This article explain's better than me what I'm trying to say. Maybe its good to know what think @LukasReschke since he are more expert in that things.

http://stackoverflow.com/questions/436411/where-is-the-best-place-to-put-script-tags-in-html-markup

**Comment by jancborchardt** [comment]
There’s also PageSpeed Insights which offers some helpful tips:
https://developers.google.com/speed/pagespeed/insights/?url=nextcloud.com

Deferring off-screen images would be one of the biggest speed gains, cutting load time in half.
